src/runtime/pkg/containerd-shim-v2/shim_management.go

```diff
@@ -307,8 +308,24 @@ func GetSandboxesStoragePath() string {
 	return "/run/vc/sbs"
 }
 
+// returns the storage path for runtime-rs created sandboxes
+// this is for consistency when using kata-runtime cli with runtime-rs
+func GetSandboxesStoragePathForRust() string {
+	return "run/kata"
+}
+
 // SocketAddress returns the address of the unix domain socket for communicating with the
 // shim management endpoint
 func SocketAddress(id string) string {
-	return fmt.Sprintf("unix://%s", filepath.Join(string(filepath.Separator), GetSandboxesStoragePath(), id, "shim-monitor.sock"))
+	// Check both runtime-rs and runtime's sandbox storage path and return the corresponding socket address
+	// Since sandbox id is unique, as long as the shim-monitor.sock exists, it should be valid
+	runtimeGoPath := fmt.Sprintf("unix://%s", filepath.Join(string(filepath.Separator), GetSandboxesStoragePath(), id, "shim-monitor.sock"))
+	runtimeRustPath := fmt.Sprintf("unix://%s", filepath.Join(string(filepath.Separator), GetSandboxesStoragePathForRust(), id, "shim-monitor.sock"))
+
+	_, err := os.Stat(runtimeGoPath)
+	if err == nil {
+		return runtimeGoPath
+	} else {
+		return runtimeRustPath
+	}
```

Comment:
Suggestion:

Change `SocketAddress` to return `(string, error)`, and:

```go
	_, err := os.Stat(runtimeGoPath)
	if err == nil {
		return runtimeGoPath, nil
	}

	_, err = os.Stat(runtimeRustPath)
	if err == nil {
		return runtimeRustPath, nil
	}

	return "", fmt.Errorf("no shim monitor socket address found")
```

Reply:
Thanks @liubin , nice suggestion, fixed.

Reply:
You are right, now the function is fixed.
1. check if rust path exists, if so, return rust path
2. if not, return go path